presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -281,7 +281,7 @@ private static long stringPositionFromEnd(Slice string, Slice substring, long in
         }
         while (foundInstances < instance);
 
-        return countCodePoints(string, 0, index) + 1;
```

Comment:
I think this behavior is intentional: see the note at the beginning here: https://prestodb.io/docs/current/functions/string.html#string-functions

Reply:
I see, but this leads to unintuitive behavior, for example

`assertFunction("STRPOS('\u4FE1\u5FF5,\u7231,\u5E0C\u671B', '\u7231')", BIGINT, 4L);` // strpos returns 4
while `assertFunction("STRRPOS('\u4FE1\u5FF5,\u7231,\u5E0C\u671B', '\u7231')", BIGINT, 2L);` // strrpos returns 2

strrpos and strpos return different result for same input (here it all matches the only \u7231 here).
And users are using substr(input, strrpos(input, delim)+1) which does not return the expected result (they expect to get the substr after the last delim)

Reply:
oh i see. Thanks for clarifying. It's definitely a bug.

It looks like the difference between strpos and strrpos is that strpos operates on a slice (Slice.indexOf()),which seems to be about bytes, so then it needs to count codepoints after getting the index. strrpos converts to utf8 strings (String.lastIndexOf()), so the index returned  has a different meaning from the one used in strpos/slices. It already refers to code points. Could use a comment to explain the difference between these two otherwise very similar looking functions, but otherwise this change looks good.

By the way, could also take the slice -> String conversion outside of the loop.

Reply:
Added comments, and also move the slice -> String conversion outside of the loop